### PR TITLE
Add some features

### PR DESCRIPTION
**Pull request by miiwu**
- feat: hightlight readme markdown checkbox
- feat: support sourcegraph, a chrome extension
- feat: support github hovercard, a chrome extension
- fix: home dashboard
- fix: repo security sidenav
- fix: repo insight forks

### CODE REVIEW

**Comment by miiwu** [comment]
- feat: hightlight readme markdown checkbox

  - before
  repo readme checkbox

  - after
  repo readme checkbox fixed


- feat: support sourcegraph, a chrome extension

  - before
  sourcegraph

  - after
  sourcegraph fixed

- feat: support github hovercard, a chrome extension

 hovercard

- fix: home dashboard

  - before
  home dashboard

  - after
  home dashboard fixed

- fix: repo security sidenav

  repo security sidenav

- fix: repo insight forks

  - before
  repo insight fork

  - after
  repo insight fork fixed

**Comment by miiwu** [comment]
Sometimes, the stylus can't change theme configuration of material-github, restart would help.
Please check it.
If caused by my chrome browser environment, please tell me.
I use stylus recently, may have no idea about it.
Thanks.

**Comment by miiwu** [comment]
@CharlieEtienne 

Your just latest commits seem to be no fix, and mine try to handle:

image

image

image

May more, could you just view my commits?

- - -

Please say something? Are these commits too bad, or something?
Or close it.

**Comment by detobel36** [comment]
Did you fix the problem in "Network" ?

Capture

(you could maybe merge `master` in you branch 😉 )

**Comment by miiwu** [comment]
> Did you fix the problem in "Network" ?

No, that is not fixed.
It may be complex for me.

> (you could maybe merge `master` in you branch 😉 )

Do you mean merge the latest update of the main to my fork?

I can't make full sense of what you mean.
Could you make a further explanation for me?

**Comment by detobel36** [comment]
> Do you mean merge the latest update of the main to my fork?
> 
> I can't make full sense of what you mean.
> Could you make a further explanation for me?

Yep, I see you did it... 😃 
If I have time, I'll look at the "network" page to fix it.
Thank you for your contribution (if this pull request is not merge, I will based myself on your work)

**Comment by miiwu** [comment]
> Yep, I see you did it... 😃 
> If I have time, I'll look at the "network" page to fix it.
> Thank you for your contribution (if this pull request is not merge, I will based myself on your work)

Oh, It's my pleasure.

**Comment by CharlieEtienne** [comment]
Hi @miiwu, sorry for the delay... I've been very busy. I see you did a lot of work here, and I need to review it. I'll try to do it today or tomorrow.

Thanks,

Charlie

**Comment by CharlieEtienne** [comment]
@detobel36 unfortunately the network graph can't be styled because it is a canvas element. We can't modify it with CSS.

File: material-github.user.css [hunk 1]
```diff
@@ -1063,5 +1063,87 @@
...
+	input[type="checkbox"],
+	input[type="checkbox"]:focus,
+	input[type="checkbox"]:hover:active,
+	input[type="checkbox"]:disabled,
+	input[type="checkbox"]:hover:disabled,
+	input[type="checkbox"]:hover:active:disabled {
+		-webkit-appearance: none !important;
+		background-color: #25272a !important;
+		border: 1px solid #484848 !important;
+		color: #eee !important;
```
Comment (CharlieEtienne) [inline material-github.user.css]: Could you use material github vars (like `--bg2` or `--selected` for example) or rgba black/white transparent nuances (ie. `rgba(0, 0, 0, .7)` instead to make it compatible with all themes ?

File: material-github.user.css [hunk 2]
```diff
@@ -1063,5 +1063,87 @@
...
+		border-top-color: var(--bg4) !important;
+	}
+	.ghh-theme-github .tooltipster-bottom .tooltipster-arrow-border {
+		border-bottom-color: var(--bg4);
+	}
+	.ghh-theme-github .tooltipster-bottom .tooltipster-arrow-background {
+		border-bottom-color: var(--bg4);
+	}
+	.ghh-state-verified {
+		color: rgb(34, 134, 58) !important;
```
Comment (CharlieEtienne) [inline material-github.user.css]: same thing here (I don't know what this line do)

**Review by CharlieEtienne — CHANGES_REQUESTED** [review]

File: material-github.user.css [hunk 3]
```diff
@@ -1117,33 +1117,84 @@
...
+		background-color: var(--bg2)!important;
+		color: rgba(255, 255, 255, 0.65) !important;
 	}
-	.ghh-theme-github .tooltipster-bottom .tooltipster-arrow-border {
-		border-bottom-color: var(--bg4);
+
+	.ghh-readme pre,
+	.ghh-issue-body pre {
+		background: var(--bg4) !important;
+		color: #ccc !important;
```
Comment (CharlieEtienne) [inline material-github.user.css]: Same thing here maybe

File: material-github.user.css [hunk 4]
```diff
@@ -1117,33 +1117,84 @@
...
 	}
-	.ghh-theme-github .tooltipster-bottom .tooltipster-arrow-background {
-		border-bottom-color: var(--bg4);
+
+	.ghh-stats span {
+		color: white!important;
 	}
+
+	.ghh-state-open {
+		color: rgb(44, 190, 78)!important;
```
Comment (CharlieEtienne) [inline material-github.user.css]: Same thing here maybe

File: material-github.user.css [hunk 5]
```diff
@@ -1117,33 +1117,84 @@
...
+		border-color: var(--bg) !important;
+		background-color: var(--bg4) !important;
+	}
+
 	.ghh-state-verified {
 		color: rgb(34, 134, 58) !important;
```
Comment (CharlieEtienne) [inline material-github.user.css]: Same thing here maybe

File: material-github.user.css (hunk 3 above)
Comment (CharlieEtienne) [reply material-github.user.css]: oops, I think this is mine, sorry

**Comment by CharlieEtienne** [comment]
Except for those, the rest seems fine! 
Next time, it would be better to make one PR per feat/fix if possible. It would be simpler to merge/review

Thanks

**Comment by miiwu** [comment]
> I've been very busy.

I should have thought of that.
I didn't mean to blame.

> Except for those, the rest seems fine!

I would do my best to make these compatible with all themes as soon as possible.

File: material-github.user.css (hunk 5 above)
Comment (miiwu) [reply material-github.user.css]: Try to fix.

File: material-github.user.css (hunk 4 above)
Comment (miiwu) [reply material-github.user.css]: Try to fix.

File: material-github.user.css (hunk 3 above)
Comment (miiwu) [reply material-github.user.css]: May be mine, haha.
Try to fix.

File: material-github.user.css (hunk 2 above)
Comment (miiwu) [reply material-github.user.css]: Try to fix.

File: material-github.user.css (hunk 1 above)
Comment (miiwu) [reply material-github.user.css]: Try to fix.

File: material-github.user.css [hunk 6]
```diff
@@ -1181,9 +1181,9 @@
 	input[type="checkbox"]:hover:disabled,
 	input[type="checkbox"]:hover:active:disabled {
 		-webkit-appearance: none !important;
-		background-color: #25272a !important;
-		border: 1px solid #484848 !important;
-		color: #eee !important;
+		background-color: rgba(37,39,42) !important; 
+		border: 1px solid rgba(72,72,72) !important;
+		color: rgba(238,238,238) !important;
```
Comment (CharlieEtienne) [inline material-github.user.css]: Sorry, that's not what I meant. You should instead use `var(--bg3)` (replace 3 with the number that applies) when it's possible, or `rgba(255, 255, 255, .7)` (replace 7 with the number that applies) in certain cases like text color sometimes. 
This is required to make the colors variable, and work with all different theme variations (Ocean, Darker, Palenight, or Custom)

**Review by CharlieEtienne — CHANGES_REQUESTED** [review]

File: material-github.user.css (hunk 6 above)
Comment (miiwu) [reply material-github.user.css]: Oh, Let me try again.

File: material-github.user.css (hunk 4 above)
Comment (miiwu) [reply material-github.user.css]: state, should be fixed

File: material-github.user.css (hunk 3 above)
Comment (miiwu) [reply material-github.user.css]: font color, maybe

File: material-github.user.css (hunk 5 above)
Comment (miiwu) [reply material-github.user.css]: state, should be fixed

File: material-github.user.css [hunk 7]
```diff
@@ -1080,8 +1080,282 @@
...
+
+	.ghh-issue-body {
+		background-color: var(--bg2)!important;
+		color: rgba(255, 255, 255, 0.65) !important;
+	}
+
+	.ghh-readme pre,
+	.ghh-issue-body pre {
+		background: var(--bg4) !important;
+		color: rgba(204,204,204) !important;
```
Comment (CharlieEtienne) [inline material-github.user.css]: rgba should take 4 parameters, I think you could set it to `rgba(255,255,255, .8)`

File: material-github.user.css [hunk 8]
```diff
@@ -1080,8 +1080,282 @@
...
+		height: 0 !important;
+		padding-top: 22px !important;
+	}
+
+	/* Repo, Releases, Releases */
+	.release-main-section {
+		border-left: 2px solid var(--bg5)!important;
+	}
+	.select-menu-modal position-absolute text-md-left .SelectMenu-list {
+		background-color: var(--bg)!important;
```
Comment (CharlieEtienne) [inline material-github.user.css]: `var(--bg)` doesn't exist

File: material-github.user.css [hunk 9]
```diff
@@ -1080,8 +1080,282 @@
...
+	}
+	.select-menu-modal position-absolute text-md-left .select-menu-list {
+		background-color: var(--bg);
```
Comment (CharlieEtienne) [inline material-github.user.css]: `var(--bg)` doesn't exist

File: material-github.user.css (hunk 7 above)
Comment (CharlieEtienne) [reply material-github.user.css]: actually, please set it to `color: rgba(255, 255, 255, calc(0.8 * /*[[text_brightness]]*/)) !important;`
This would make it compatible with text brightness selector.
Thanks!

File: material-github.user.css [hunk 10]
```diff
@@ -1080,8 +1080,282 @@
...
+	input[type="checkbox"]:focus,
+	input[type="checkbox"]:hover:active,
+	input[type="checkbox"]:disabled,
+	input[type="checkbox"]:hover:disabled,
+	input[type="checkbox"]:hover:active:disabled {
+		-webkit-appearance: none !important;
+		border: 1px solid var(--bg5) !important;
+		height: 1em !important;
+		width: 1em !important;
+	}
```
Comment (CharlieEtienne) [inline material-github.user.css:1205]: Did you just remove background-color and color properties ? :D

File: material-github.user.css [hunk 11]
```diff
@@ -1080,8 +1080,282 @@
...
+
+	/* Home, Explore repositories */
+	.py-2.my-2.border-bottom {
+		border-color: var(--bg2) !important;
+	}
+
+	/* Profile, Repositories, Type & Language */
+	label.SelectMenu-item[aria-checked=true] {
+		color: rgba(255, 255, 255, 0.7)!important;
```
Comment (CharlieEtienne) [inline material-github.user.css]: could you please set it to `color: rgba(255, 255, 255, calc(0.7 * /*[[text_brightness]]*/)) !important;`
This would make it compatible with text brightness selector.

File: material-github.user.css [hunk 12]
```diff
@@ -1080,8 +1080,282 @@
...
+	}
+
+	/* Repo, Header, recent pushes */
+	.Box-row--yellow {
+		background-color: var(--bg3)!important;
+		color: var(--bg5)!important;
+	}
+	.flex-auto {
+		color: rgba(255, 255, 255, 0.7) !important
+	}
```
Comment (CharlieEtienne) [inline material-github.user.css]: What does this thing ? The class name suggests it would affect a lot of elements  (maybe I'm wrong)

If we keep it, we should use `color: rgba(255, 255, 255, calc(0.7 * /*[[text_brightness]]*/)) !important;` to make it compatible with text brightness selector.

File: material-github.user.css [hunk 13]
```diff
@@ -1080,8 +1080,282 @@
...
+
 	.review-comment .is-comment-editing {
 	    background-color: var(--bg1);
-	    border: var(--bg1);
+		border: var(--bg1);
+	}
+	
+	/* Header */
+	.Header-link {
+		color: rgba(255, 255, 255, 0.6) !important;
```
Comment (CharlieEtienne) [inline material-github.user.css]: could you please set it to `color: rgba(255, 255, 255, calc(0.6 * /*[[text_brightness]]*/)) !important;`
This would make it compatible with text brightness selector.

File: material-github.user.css [hunk 14]
```diff
@@ -1080,8 +1080,282 @@
...
+	.Header-link:focus,
+	.Header-link:hover {
+		color: var(--selected) !important;
+	}
+
+	/* Home, Repositories */
+	button.width-full.text-left.btn-link.f6.muted-link,
+	h2.hide-sm.hide-md.f5.mb-1.border-top.pt-3 {
+		border-top: 1px solid var(--bg4)!important;
+		color: rgba(255, 255, 255, 0.6) !important;
```
Comment (CharlieEtienne) [inline material-github.user.css]: could you please set it to `color: rgba(255, 255, 255, calc(0.6 * /*[[text_brightness]]*/)) !important;`
This would make it compatible with text brightness selector.

File: material-github.user.css [hunk 15]
```diff
@@ -1080,8 +1080,282 @@
...
+	li.Box-row.d-flex.p-2,
+	li.d-flex.border-top.p-2.position-relative {
+		border-top: 1px solid var(--bg3)!important;
+	}
+	li.Box-row.d-flex.p-2:first-of-type {
+		border-top-color: transparent!important;
+	}
+	.js-details-target.js-show-more-recent-items {
+		background-color: var(--bg2)!important;
+		color: rgba(255, 255, 255, 0.6) !important;
```
Comment (CharlieEtienne) [inline material-github.user.css]: could you please set it to `color: rgba(255, 255, 255, calc(0.6 * /*[[text_brightness]]*/)) !important;`
This would make it compatible with text brightness selector.

File: material-github.user.css (hunk 10 above)
Comment (miiwu) [reply material-github.user.css:1205]: I see that would have no effect superficially, just ill-considered maybe.

File: material-github.user.css (hunk 8 above)
Comment (miiwu) [reply material-github.user.css]: Oh, well, that's awkward.
My brain must have been confused.

File: material-github.user.css (hunk 12 above)
Comment (miiwu) [reply material-github.user.css]: A further check needed.
I would conduct as soon as possible.

File: material-github.user.css (hunk 7 above)
Comment (miiwu) [reply material-github.user.css]: This was omissive last night.
I would check if have rgba without taking the 4th parameters as soon as possible.

**Comment by miiwu** [comment]
The Github keeps changing seemingly, so i would stay these changes and fix them.

File: material-github.user.css (hunk 12 above)
Comment (miiwu) [reply material-github.user.css]: Maybe 'recent pushes' disappear...
It appears suddently many days ago.

**Comment by CharlieEtienne** [comment]
@miiwu I fixed white "T" images in Insights > Forks in https://github.com/CharlieEtienne/material-github/commit/28f346303d40e065efbc7a420c626919e91fc763 with a base64 background-image. This way we can keep it instead of hiding it.

**Comment by miiwu** [comment]
> @miiwu I fixed white "T" images in Insights > Forks in [28f3463](https://github.com/CharlieEtienne/material-github/commit/28f346303d40e065efbc7a420c626919e91fc763) with a base64 background-image. This way we can keep it instead of hiding it.

Awesome, would make a big step!
My work is not very well due to cognition.

- - -
Attach a preview would help, thanks. 😁

**Comment by miiwu** [comment]
`Changes Request`, is there any conflicts yet?

- - -
If not, i could delete those, have had a better solution?
And, i might not open my pc these days, you could do these for me?

**Review by CharlieEtienne — APPROVED** [review]

**Comment by CharlieEtienne** [comment]
@miiwu this PR becomes a mess... I'm ready to accept and merge some changes like extensions support, fix for Security nav menu, but I can't review these changes here anymore. 
Would you open one separate PR for each feat/fix?

Thanks a lot,

Charlie

**Comment by miiwu** [comment]
> @miiwu this PR becomes a mess...

Yes, i agree.

> Would you open one separate PR for each feat/fix?

I would separate these as soon as possible.

Each PR for each?
That's a lot, maybe.

**Comment by CharlieEtienne** [comment]
@miiwu I don't know but I'm losing the count:

It started with 6

> - feat: hightlight readme markdown checkbox
> - feat: support sourcegraph, a chrome extension
> - feat: support github hovercard, a chrome extension
> - fix: home dashboard
> - fix: repo security sidenav
> - fix: repo insight forks

but now I don't know where we are.

You could group some if you feel it is still readable but the simpler the PR is, the sooner it will be merged :)

**Comment by miiwu** [comment]
> You could group some if you feel it is still readable but the simpler the PR is, the sooner it will be merged :)

Yes, i would,  hahaha.
